[UXE-3212] feat: add azion config file docs (#971)
aziontech/docs: src/content/docs/en/pages/devtools/cli/azion-cli/config/azion-config-js.mdx

---
title: azion.config.js file

> description: Learn how to customize your application build process using the azion.config.js file. This documentation provides detailed explanations of the configuration properties.

Reviewer: ```suggestion
description: Learn how to customize your application's build process using the azion.config.js file. This documentation provides detailed explanations of the configuration properties.
```

meta_tags: azion, configuration, javascript, build process
namespace: documentation_devtools_configs_azion_config_js
permalink: /documentation/devtools/cli/configs/azion-config-js/
menu_namespace: cliMenuAlpha
---

> The `azion.config.js` file is a configuration file created during the build of the application, based on the chosen preset. Each preset presents a set of default settings. These settings can be replaced by the user. This file is the source of truth for the configuration. In case it's deleted, when the build process is triggered again, the default configuration will be recreated.

Reviewer: ```suggestion
The azion.config.js file is a configuration file created during the application's build process, based on the chosen preset, and serves as the source of truth for the configuration. Each preset provides a set of default settings, which can be replaced by the user. If it’s deleted, the default configuration will be recreated the next time the build process is triggered.
```

The following tables explain the properties of this configuration file.

---

## origin

| Property | Description | Values | Required |
| --- | --- | - | - |

> | name | The name of the origin. |  | Yes |

Reviewer: ```suggestion
| name | The name of the origin |  | Yes |
```

> | type | The type of the origin. In this case, it's an object storage.  |  Possible values: `single_origin` (default)` load_balancer`,`live_ingest` or `object_storage` | Required |

Reviewer: ```suggestion
| type | The type of the origin. In this case, it's an object storage.  |  Possible values: `single_origin` (*default*)`, load_balancer`,`live_ingest`, or `object_storage` | Required |
```

> | type | The type of the origin. In this case, it's an object storage.  |  Possible values: `single_origin` (default)` load_balancer`,`live_ingest` or `object_storage` | Required |

Reviewer: ```suggestion
| type | The type of the origin. In this case, it's an object storage  |  Possible values: `single_origin` (default)` load_balancer`,`live_ingest` or `object_storage` | Required |
```

> | bucket | The name of the bucket in the object storage. |  | Yes for object_storage type origins. |

Reviewer: ```suggestion
| bucket | The name of the bucket in the object storage. |  | Yes, for object_storage type origins. |
```

| prefix | The prefix for the objects in the bucket. |   | No |

> [For further information, access Azion API specification](https://api.azion.com/#adbc0ab3-8f27-47e2-a8fe-428c7fa93250)

Reviewer: ```suggestion
For further information, [access Azion API specification](https://api.azion.com/#adbc0ab3-8f27-47e2-a8fe-428c7fa93250)
```

---

## cache

| Property | Description |  Required |
| --- | --- | --- |

> | name | Name of your cache setting. | Yes|
> | stale | A boolean indicating whether stale content can be served. | |
> | queryStringSort | Considers objects with the same query strings as the same cached file regardless of the order of the fields. | |
> | methods | An object specifying which HTTP methods should be cached. | |
> | browser | An object specifying the maximum age of the cache in the browser. | |
> | edge | An object specifying the maximum age of the cache at the edge. | |

Reviewer: ```suggestion
| name | Name of your cache setting | Yes |
| stale | A boolean indicating whether stale content can be served | |
| queryStringSort | Considers objects with the same query strings as the same cached file regardless of the order of the fields | |
| methods | An object specifying which HTTP methods should be cached | |
| browser | An object specifying the maximum age of the cache in the browser | |
| edge | An object specifying the maximum age of the cache at the edge | |
```

Reviewer: You can add "No" in the Required column for non-required items. To align with the first table.

[For further information, access Azion API specification](https://api.azion.com/#6a337475-b474-416d-b710-ae11ac88b189)

---

## rules

> | Property | Description |   Required |
> | --- | --- | --- |
> | request | An array of request rules. Each rule has properties like name, match, cache, rewrite, setCookie, setHeaders, forwardCookies, setOrigin, deliver, and runFunction. | No |
> | name | The name of the rule. | Yes |
> | match | The pattern to match for the rule. | |
> | cache | The name of the cache to use for the rule. | |
> | rewrite | An object specifying the match and set properties for rewriting the request. | |
> | setCookie | A string specifying the Set-Cookie header. | |
> | setHeaders | A string specifying the headers to set. | |
> | forwardCookies | A boolean indicating whether cookies should be forwarded. | |
> | setOrigin | An object specifying the name and type of the origin to set. | |
> | deliver | A boolean indicating whether the content should be delivered. | |
> | runFunction | An object specifying the path of the function to run. | |

Reviewer: ```suggestion
| Property | Description | Required |
| --- | --- | --- |
| request | An array of request rules. Each rule has properties like name, match, cache, rewrite, setCookie, setHeaders, forwardCookies, setOrigin, deliver, and runFunction | No |
| name | The name of the rule | Yes |
| match | The pattern to match for the rule | |
| cache | The name of the cache to use for the rule | |
| rewrite | An object specifying the match and set properties for rewriting the request | |
| setCookie | A string specifying the Set-Cookie header | |
| setHeaders | A string specifying the headers to set | |
| forwardCookies | A boolean indicating whether cookies should be forwarded | |
| setOrigin | An object specifying the name and type of the origin to set | |
| deliver | A boolean indicating whether the content should be delivered | |
| runFunction | An object specifying the path of the function to run | |
```

Reviewer: Same about the Required column

> [For further information, access Azion API specification](https://api.azion.com/#42279c76-58b3-4ce5-9852-1bce43c89524)

Reviewer: ```suggestion
For further information, [access Azion API specification](https://api.azion.com/#42279c76-58b3-4ce5-9852-1bce43c89524)
```
